File: ruby_small_problems/easy_7/7.rb
# takes array of integers as argument
# multiplies all the numbers togehter
# divides the result by the number of entries in the array
# prints the result rounded to 3 decimal places
# assume the array is non-empty
# 
# .reduce(:*)
# / arr.size.to_f

def show_multiplicative_average(arr)
  result = (arr.reduce(:*) / arr.size.to_f)
  puts format("The result is %.3f", result)
end

show_multiplicative_average([3, 5])

show_multiplicative_average([6])

show_multiplicative_average([2, 5, 7, 11, 13, 17])

File: 210/exercises/small_problems/easy_3/7.js
function runningTotal(arr) {
  if (arr.length === 0) {
    return arr; 
  }

  let result = [arr[0]];
  let i;
  for (i = 1; i < arr.length; i += 1) {
    result.push(result[result.length - 1] + arr[i]); 
  }
  return result;
}

console.log(runningTotal([2, 5, 13]));             // [2, 7, 20]
console.log(runningTotal([14, 11, 7, 15, 20]));    // [14, 25, 32, 47, 67]
console.log(runningTotal([3]));                    // [3]
console.log(runningTotal([]));                     // []
